qiskit/transpiler/passmanager.py

```diff
@@ -412,9 +408,9 @@ def __init__(self, stages: Optional[Iterable[str]] = None, **kwargs) -> None:
             setattr(self, stage, pm)
 
     def _validate_stages(self, stages: Iterable[str]) -> None:
-        invalid_stages = [
-            stage for stage in stages if self.invalid_stage_regex.search(stage) is not None
-        ]
+        if not isinstance(stages, Iterable) or any([not isinstance(s, str) for s in stages]):
+            raise TypeError("StagedPassManager stages must be of type Iterable[str] or None")
```

Comment:
This sort of testing is generally a bit too much for a library - if the user is clearly violating the documented input types, we don't need to be responding for giving them a pretty error message (since it slows down the code for people following the instructions).

Reply:
Thanks for bringin up the list comprehension! you are absolutely correct about it